CRM/Contact/Import/Parser/Contact.php

```diff
@@ -1050,29 +1050,27 @@ private function getSuccessMessage(): string {
    *
    * @param array $params
    * @param int|null $extIDMatch
-   * @param int|null $dedupeRuleID
+   * @param int|string $dedupeRuleID
    *
    * @return int|null
    *   IDs of a possible.
    *
    * @throws \CRM_Core_Exception
    * @throws \CiviCRM_API3_Exception
    */
-  protected function getPossibleContactMatch(array $params, ?int $extIDMatch, ?int $dedupeRuleID): ?int {
-    $checkParams = ['check_permissions' => FALSE, 'match' => $params, 'dedupe_rule_id' => $dedupeRuleID];
-    $possibleMatches = civicrm_api3('Contact', 'duplicatecheck', $checkParams);
+  protected function getPossibleContactMatch(array $params, ?int $extIDMatch, $dedupeRuleID): ?int {
```

Comment:
I would suggest one further refactor, to change this name to require a string for the 3rd param and take `$dedupeRuleName` instead of `$dedupeRuleID`. This function is only called from one place (a quickform) so that form element for selecting a rule can simply be changed to a list keyed by name instead of id.

Noe a blocker tho.

Reply:
yeah I went back & forth on that - but the user has selected an id at this point - so it has to be converted 'somewhere'